packages/xchain-bitcoincash/src/client.ts

```diff
@@ -0,0 +1,331 @@
+  }
+
+  /**
+   * Set/update a new phrase.
+   *
+   * @param {string} phrase A new phrase.
+   * @returns {Address} The address from the given phrase
+   *
+   * @throws {"Invalid phrase"}
+   * Thrown if the given phase is invalid.
+   */
+  setPhrase = (phrase: string): Address => {
+    if (validatePhrase(phrase)) {
+      this.phrase = phrase
+      const address = this.getAddress()
+      return address
+    } else {
+      throw new Error('Invalid phrase')
+    }
+  }
+
+  /**
+   * Purge client.
+   *
+   * @returns {void}
+   */
+  purgeClient = (): void => {
+    this.phrase = ''
+  }
+
+  /**
+   * Set/update the current network.
+   *
+   * @param {Network} network `mainnet` or `testnet`.
+   * @returns {void}
+   *
+   * @throws {"Network must be provided"}
+   * Thrown if network has not been set before.
+   */
+  setNetwork = (network: Network): void => {
+    if (!network) {
+      throw new Error('Network must be provided')
+    } else {
+      this.network = network
+    }
+  }
+
+  /**
+   * Get the current network.
+   *
+   * @returns {Network} The current network. (`mainnet` or `testnet`)
+   */
+  getNetwork = (): Network => {
+    return this.network
+  }
+
+  /**
+   * Get DerivePath
+   *
+   * @returns {string} The bitcoin cash derivation path based on the network.
+   */
+  derivePath(): string {
+    const { testnet, mainnet } = utils.getDerivePath()
+    return utils.isTestnet(this.network) ? testnet : mainnet
+  }
+
+  /**
+   * Get the explorer url.
+   *
+   * @returns {string} The explorer url based on the network.
+   */
+  getExplorerUrl = (): string => {
+    const networkPath = utils.isTestnet(this.network) ? 'tbch' : 'bch'
+    return `https://explorer.bitcoin.com/${networkPath}`
+  }
+
+  /**
+   * Get the explorer url for the given address.
+   *
+   * @param {Address} address
+   * @returns {string} The explorer url for the given address based on the network.
+   */
+  getExplorerAddressUrl = (address: Address): string => {
+    return `${this.getExplorerUrl()}/address/${address}`
+  }
+
+  /**
+   * Get the explorer url for the given transaction id.
+   *
+   * @param {string} txID The transaction id
+   * @returns {string} The explorer url for the given transaction id based on the network.
+   */
+  getExplorerTxUrl = (txID: string): string => {
+    return `${this.getExplorerUrl()}/tx/${txID}`
+  }
+
+  /**
+   * @private
+   * Get private key.
+   *
+   * Private function to get keyPair from the this.phrase
+   *
+   * @param {string} phrase The phrase to be used for generating privkey
+   * @returns {PrivateKey} The privkey generated from the given phrase
+   * */
+  private getPrivateKey = (phrase: string): bitcash.PrivateKey => {
+    const derive_path = this.derivePath()
+    const mnemonic = new Mnemonic(phrase)
+    const hdPrivKey: bitcash.HDPrivateKey = mnemonic.toHDPrivateKey().derive(derive_path)
+
+    return bitcash.PrivateKey.fromObject(hdPrivKey.privateKey)
```

Comment:
Pls check original sources of `mnemonic` / `bitcash` libs if any of this functions can fail by throwing an error.